src/muqsit/arenagenerator/layout/chunk/PreloadedChunk.php

```diff
@@ -4,90 +4,66 @@
 
 namespace muqsit\arenagenerator\layout\chunk;
 
+use pocketmine\block\tile\Tile;
 use pocketmine\block\tile\TileFactory;
-use pocketmine\entity\EntityFactory;
-use pocketmine\nbt\tag\CompoundTag;
-use pocketmine\world\format\BiomeArray;
 use pocketmine\world\format\Chunk;
-use pocketmine\world\format\HeightArray;
 use pocketmine\world\format\SubChunk;
 use pocketmine\world\World;
 
 final class PreloadedChunk{
 
-	public static function fromChunk(Chunk $chunk) : self{
+    public static function fromChunk(Chunk $chunk) : self{
 		$chunk = clone $chunk;
 		return new self(
-			$chunk->getSubChunks()->toArray(),
-			$chunk->getNBTentities(),
-			$chunk->getNBTtiles(),
-			$chunk->getBiomeIdArray(),
+			$chunk->getSubChunks(),
+			$chunk->getTiles(),
 			$chunk->getHeightMapArray(),
 			$chunk->isLightPopulated()
 		);
 	}
 
-	/** @var SubChunk[] */
+    /** @var SubChunk[] */
 	private array $sub_chunks;
 
-	/** @var CompoundTag[] */
-	private array $entities;
-
-	/** @var CompoundTag[] */
+    /** @var Tile[] */
 	private array $tiles;
 
-	private string $biome_ids;
-
-	/** @var int[] */
-	private array $height_map;
+    /** @var int[] */
+    private array $height_map;
 
-	private ?bool $light_populated;
+    private ?bool $light_populated;
 
-	/**
+    /**
 	 * @param SubChunk[] $sub_chunks
-	 * @param CompoundTag[] $entities
-	 * @param CompoundTag[] $tiles
-	 * @param string $biome_ids
+	 * @param Tile[] $tiles
 	 * @param int[] $height_map
 	 * @param bool|null $light_populated
 	 */
-	public function __construct(array $sub_chunks, array $entities, array $tiles, string $biome_ids, array $height_map, ?bool $light_populated){
+	public function __construct(array $sub_chunks, array $tiles, array $height_map, ?bool $light_populated){
```

Comment:
Existing entities must be transferred onto copied arenas (this change will prevent stuff like zombies in a region or hung paintings to be lost during copying).